recipe/install_nvcc.sh

```diff
@@ -2,61 +2,75 @@
 
 set -xeuo pipefail
 
-# Default to using `nvcc` to specify `CUDA_HOME`.
-if [ -z ${CUDA_HOME+x} ]
-then
-    CUDA_HOME="$(dirname $(dirname $(which nvcc)))"
-fi
-
 # Set `CUDA_HOME` in an activation script.
 mkdir -p "${PREFIX}/etc/conda/activate.d"
-cat > "${PREFIX}/etc/conda/activate.d/${PKG_NAME}_activate.sh" <<EOF
+cat >"${PREFIX}/etc/conda/activate.d/${PKG_NAME}_activate.sh" <<EOF
 #!/bin/bash
 
-if [[ ! -d "${CUDA_HOME}" ]]
+CUDA_HOME_UNSET=0
+
+# Default to using \$(cuda-gdb) to specify \$(CUDA_HOME).
+if [ -z \${CUDA_HOME+x} ]
 then
-    echo "Cannot find: ${CUDA_HOME}"
+    CUDA_HOME="\$(dirname \$(dirname \$(which cuda-gdb)))"
+    CUDA_HOME_UNSET=1
+fi
+
+if [[ ! -d "\${CUDA_HOME}" ]]
+then
+    echo "Cannot find: \${CUDA_HOME}"
     exit 1
 fi
-if [[ ! -f "${CUDA_HOME}/lib64/stubs/libcuda.so" ]]
+if [[ ! -f "\${CUDA_HOME}/lib64/stubs/libcuda.so" ]]
 then
-    echo "Cannot find: ${CUDA_HOME}/lib64/stubs/libcuda.so"
+    echo "Cannot find: \${CUDA_HOME}/lib64/stubs/libcuda.so"
     exit 1
 fi
-grep "CUDA Version ${PKG_VERSION}" ${CUDA_HOME}/version.txt &>/dev/null
-if [[ $? -ne 0 ]]
+grep "CUDA Version ${PKG_VERSION}" \${CUDA_HOME}/version.txt &>/dev/null
+if [[ \$? -ne 0 ]]
```

Comment:
Note that conda activate scripts are pre-processed. So `$?` may not contain the return status of the `grep` command as intended. As a fix, use:
```
if grep -q "CUDA Version ${PKG_VERSION}" \${CUDA_HOME}/version.txt; then
    :
else
    echo "Version of installed CUDA in \${CUDA_HOME} didn't match package ${PKG_VERSION} (\$?)"
    exit 1
fi
```